force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,55 @@
+trigger OpportunityTrigger on Opportunity (before insert, before update, before delete, after update, after delete) {
+
+    switch on Trigger.operationType {
+        when AFTER_UPDATE {
+            for(Opportunity opp : Trigger.New) {
+                if(opp.Amount < 5000) {
+                    Trigger.new[0].addError('Opportunity amount must be greater than 5000');
```

Comment:
Try to throw the validation error before the update not after, after it's already too late